File: 0179-largest-number/0179-largest-number.cpp
class Solution {
public:
    static bool cmp(string &a, string &b){   //compare a = 3, b = 30
        return a + b > b + a;               // 330 < 303
    }
    string largestNumber(vector<int>& nums) {
        vector<string>temp;
        for(int n : nums) temp.push_back(to_string(n));
        sort(temp.begin(), temp.end(), cmp);

        if(temp[0] == "0") return "0";

        string res = "";

        for(int i=0; i<temp.size(); i++){
            res = res + temp[i];
        }

        return res;
    }
};

File: 0179-largest-number/0179-largest-number.py
class Solution:
    from functools import cmp_to_key
    def largestNumber(self, nums: List[int]) -> str:
        arr = list(map(str,nums))
        
        arr.sort(
            key=cmp_to_key(
                lambda a,b : -1 if a+b > b+a else (1 if a+b < b+a else 0)
            )
        )

        return "0" if arr[0] == "0" else "".join(arr)